src/Apps/W1/PEPPOL/App/src/Common/PEPPOL30.Codeunit.al

```diff
@@ -0,0 +1,1115 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace Microsoft.Peppol;
+
+using Microsoft.Finance.VAT.Calculation;
+using Microsoft.Finance.VAT.Setup;
+using Microsoft.Foundation.Attachment;
+using Microsoft.Sales.Document;
+using Microsoft.Sales.History;
+using Microsoft.Service.Document;
+
+codeunit 37200 "PEPPOL30" implements "PEPPOL Attachment Handler"
```

Comment:
This codeunit is too big and is implementing too many intefraces...
It is hard to understand the code if it is used directly, without an interface. Since it is not internal, people may decide to use it directly. 

Consider breaking into multiple codeunits, there can be one single codeunit that is passing a reference to a smaller interface/codeunit e.g. Peppol30.GetDocumentInfoProvider().GetDocumentNo or something like this, or we could have many small codeunits without a parent. Alternative is a Subtype=Temporary table design.